eng/scripts/Verify-PreviewDependency.ps1

```diff
@@ -0,0 +1,17 @@
+[CmdletBinding()]
+param (
+  [Parameter(Mandatory = $true)]
+  [string]$PackagePath
+)
+
+$packagePaths = Get-ChildItem -Path "$PackagePath/*/package.json"
+
+foreach ($packagePath in $packagePaths) {
+  $pkgContent = Get-Content $packagePath | ConvertFrom-Json
+  foreach ($dependency in $pkgContent.dependencies.PSObject.properties) {
+    $depVer = $dependency.Value
+    if (($depVer -match "\D0.\d.\d") -or ($depVer -match "\D\d.\d.\d-\w")) {
```

Comment:
You might want to use our SemVer class under eng/common to parse the versions. https://github.com/Azure/azure-sdk-for-js/blob/main/eng/common/scripts/SemVer.ps1

Reply:
Agreed, it helps to avoid edge cases that we've found in the past and baked into the SemVer class.